pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -103,9 +105,48 @@ func freeACEId(candidate int32) {
 	}
 }
 
+// Returns name to use for ipset matching a given domain name
+// (ACE match of type "host").
+// Needs to ensure that the ipset name doesn't exceed the length
+// limit of 31 characters imposed by netfilter.
+func hostIpsetName(hostname string) string {
+	maxLen := ipsetNameLenLimit - 5 // leave 5 characters for "ipvX."
+	if len(hostname) < maxLen {
+		return hostname
+	}
+	const (
+		// Minimum number of characters the hash should contain
+		// to bring the probability of collision down to an acceptable level.
+		hashMinLen = 8
+		// Separator between a hostname suffix and a hash-generated prefix.
+		sep = "#"
+	)
+	// Function tries to keep some suffix from the original host name
+	// (to keep the ipset name human-readable and traceable to its source)
+	// and replaces only the remaining subdomains with a hash value.
+	domains := strings.Split(hostname, ".")
```

Comment:
Nit: in DNS the things between the dots are called labels.

Reply:
Fixed